external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1546,6 +1563,9 @@ fn parse_sequence(context: &mut Context) -> Result<Sequence, Box<Diagnostic>> {
                 seq.push(item);
                 last_semicolon_loc = Some(current_token_loc(context.tokens));
                 if let Err(diag) = consume_token(context.tokens, Tok::Semicolon) {
+                    if continue_sequence_after_error(context, diag.as_ref().clone()) {
```

Comment:
I'll leave for @cgswords to review, but personally, I would love to see some exploration on using newline information. I think it feels potentially a more "general" solution, but who knows
e.g.
```
x.foo
//  ^
//  CURSOR HERE
x.bar();
```
just looking for let doesn't help here but seeing that the next token is on another line helps a lot.